### DISCUSSION TITLE
Requested Package Interval (RPI) out of range

### DISCUSSION THREAD

**Discussion by ramezanifar** [EIPStackGroup/OpENer#432] [Q&A] [answered]
Theoretically in an assembly, the RPI can be set as low as 0.2 ms and as large as 750.0 ms as indicated in the CIP Module properties, in RSLogix5000 software.
But I can only use an RPI equal or greater than 10ms.
If I use a lower value, I get this error on the PLC side:
`(Code 16#0111) Requested Packet Interval (RPI) out of range.`
Is there a setting in any of the files that limits the RPI to 10ms? 
Thank you inadvance

**Comment by MartinMelikMerkumians** [comment, marked as answer]
Its indirectly based on the select handler loop timer in generic_networkhandler.c. less than 10ms are not working for the select function as far as I know, the others which could handle faster times are not portable
